src/bentoml/_internal/context.py

```diff
@@ -88,20 +88,25 @@ def from_http(request: "starlette.requests.Request") -> "InferenceApiContext":
 
         return InferenceApiContext(request_ctx, response_ctx)
 
-    @attr.define
     class RequestContext:
         metadata: Metadata
         headers: Metadata
+        http: starlette.requests.Request | None
```

Comment:
Should we restructure the context to have the `http` and `grpc` group at the top level, so the syntax makes more syntactical sense? The downside is that we have to deal with backward compatibility.
- `ctx.http.request`
- `ctx.http.headers`
- `ctx.grpc.metadata`

Reply:
I think we should, as #3511 adds a `ctx.grpc` field.

Compatibility is pretty light for this if I'm not missing anything.